pkg/cmd/kafka/create/create.go

```diff
@@ -212,10 +194,37 @@ func runCreate(opts *options) error {
 			return nil
 		}
 
-		userInstanceType, err = accountmgmtutil.GetUserSupportedInstanceType(f.Context, &constants.Kafka.Ams, conn)
+		err = ValidateBillingModel(opts.billingModel)
+		if err != nil {
+			return err
+		}
+
+		userInstanceType, err = accountmgmtutil.GetUserSupportedInstanceType(f.Context, &constants.Kafka.Ams, conn, opts.billingModel)
 		if err != nil || userInstanceType == nil {
 			return f.Localizer.MustLocalizeError("kafka.create.error.userInstanceType.notFound")
 		}
+
+		if opts.billingModel == "marketplace" {
+
+			validMarketplaces, newErr := accountmgmtutil.GetValidMarketplaces(userInstanceType)
```

Comment:
Does this make call to AMS? I think we need to deal with all of marketplace stuff in account management package. 
Command itself should not have any logic for it - especially when we would need the same for future services as well. All we want to get is one singular quota object from accountmanagement package and our input should be billing type, marketplaceType, marketplaceAccountId